src/mscorlib/src/System/Environment.cs

```diff
@@ -1052,8 +1086,27 @@ public static IDictionary GetEnvironmentVariables()
                 root[0]++;
             }
             return result;
+#else // PLATFORM_UNIX
+            // Use reflection to access the unix implementation in System.IO.FileSystem.DriveInfo.dll
+            Type driveInfoType = Type.GetType(
+                "System.IO.DriveInfo, System.IO.FileSystem.DriveInfo, Version=4.0.0.0, Culture=neutral, PublicKeyToken=" + AssemblyRef.MicrosoftPublicKeyToken,
```

Comment:
If we end up going with reflection for this, it maybe be a good idea to use the same reflection forwarder accross all platforms.